pyinfra/api/arguments.py

```diff
@@ -71,51 +84,81 @@ def generate_env(config, value):
     "_shell_executable": {
         "description": "The shell to use. Defaults to ``sh`` (Unix) or ``cmd`` (Windows).",
         "default": lambda config: config.SHELL,
+        "type": str,
     },
     "_chdir": {
         "description": "Directory to switch to before executing the command.",
+        "type": str,
     },
     "_env": {
         "description": "Dictionary of environment variables to set.",
         "handler": generate_env,
+        "type": Mapping[str, str],
     },
     "_success_exit_codes": {
         "description": "List of exit codes to consider a success.",
         "default": lambda config: [0],
+        "type": Iterable[int],
+    },
+    "_timeout": {
+        "description": "Timeout for *each* command executed during the operation.",
+        "type": int,
+    },  # TODO not sure
+    "_get_pty": {
+        "description": "Whether to get a pseudoTTY when executing any commands.",
+        "type": bool,
+    },
+    "_stdin": {
+        "description": "String or buffer to send to the stdin of any commands.",
+        "type": str,  # TODO don't know the type for buffer
```

Comment:
This currently has to be a string, list or tuple object, although makes me think this should accept IO-like objects too! For now this should cover it, I'll add a new issue to track reading stdin from IO like objects.

```suggestion
        "type": Union[str, list, tuple],
```